Migrate security page

yakhinvadim opened the pull request:
## Summary

This PR migrates the Security tree from ent-search.

![Screen Cast 2021-01-29 at 10 37 46 AM](https://user-images.githubusercontent.com/11838280/106288331-47518d80-621e-11eb-9b8f-098ab4b43457.gif)

Public folder coverage:
![image](https://user-images.githubusercontent.com/11838280/106289782-11ada400-6220-11eb-8a60-fb9a2e27d8ea.png)

Server folder coverage:
![image](https://user-images.githubusercontent.com/11838280/106290210-96002700-6220-11eb-94c5-2ff3d8d1b037.png)


### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/)
* Commit: 988149031c7cfe5c4822134c5493546f1bcf7d5f
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/execution/node/379/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/","number":"102135","commit":"988149031c7cfe5c4822134c5493546f1bcf7d5f"}]}
PIPELINE-->

yakhinvadim commented:
I will update the PR with improved async tests after [Jason's PR](https://github.com/elastic/kibana/pull/89636) is merged.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102143/)
* Commit: 47910931e7926df740be7aa2a46f2c9f12b114c6

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/049135192e397f089adce5c50639fcd2d563d8d2) | [after](https://github.com/elastic/kibana/commit/47910931e7926df740be7aa2a46f2c9f12b114c6) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1166 | 1170 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/049135192e397f089adce5c50639fcd2d563d8d2) | [after](https://github.com/elastic/kibana/commit/47910931e7926df740be7aa2a46f2c9f12b114c6) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.8MB | 1.8MB | +19.0KB |

### History
* :broken_heart: [Build #102135](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/) failed 988149031c7cfe5c4822134c5493546f1bcf7d5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/","number":"102135","commit":"988149031c7cfe5c4822134c5493546f1bcf7d5f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102143/","number":"102143","commit":"47910931e7926df740be7aa2a46f2c9f12b114c6"}]}
PIPELINE-->

scottybollinger commented on x-pack/plugins/enterprise_search/public/applications/workplace_search/views/security/components/private_sources_table.test.tsx:
```diff
@@ -0,0 +1,57 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import '../../../../__mocks__/kea.mock';
```
I may br wrong but I don't think we need the top import anymore, thanks to some refactoring by the App Search team

scottybollinger commented on x-pack/plugins/enterprise_search/public/applications/workplace_search/views/security/security.test.tsx:
```diff
@@ -0,0 +1,115 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import '../../../__mocks__/kea.mock';
```
I may br wrong but I don't think we need the top import anymore, thanks to some refactoring by the App Search team

scottybollinger reviewed (approved):
Couple of minor nits but looks great otherwise!

yakhinvadim commented:
@scottybollinger Thanks for the review!
I addressed feedback and added a couple more commits.
- `Use nextTick instead of awaiting for promises` implements changes from Jason's PR: https://github.com/elastic/kibana/pull/89636
- `Update logic tests to use new mockHelpers` implements changes from Constance's PR: https://github.com/elastic/kibana/pull/88494

scottybollinger reviewed (approved):
🎉

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102265/)
* Commit: 733695e0b18c2fc5b3ce5714148da6a197afb3c1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/049135192e397f089adce5c50639fcd2d563d8d2) | [after](https://github.com/elastic/kibana/commit/733695e0b18c2fc5b3ce5714148da6a197afb3c1) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1166 | 1170 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/049135192e397f089adce5c50639fcd2d563d8d2) | [after](https://github.com/elastic/kibana/commit/733695e0b18c2fc5b3ce5714148da6a197afb3c1) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.8MB | 1.8MB | +19.0KB |

### History
* :green_heart: [Build #102143](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102143/) succeeded 47910931e7926df740be7aa2a46f2c9f12b114c6
* :broken_heart: [Build #102135](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/) failed 988149031c7cfe5c4822134c5493546f1bcf7d5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102135/","number":"102135","commit":"988149031c7cfe5c4822134c5493546f1bcf7d5f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102143/","number":"102143","commit":"47910931e7926df740be7aa2a46f2c9f12b114c6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102265/","number":"102265","commit":"733695e0b18c2fc5b3ce5714148da6a197afb3c1"}]}
PIPELINE-->
